pkg/pull/pull.go

```diff
@@ -696,7 +747,7 @@ func writeDownstreams(options PullOptions, overlaysDir string, m *midstream.Mids
 		}
 
 		for _, mid := range helmMidstreams {
-			helmMidstream := mid
+			helmMidstream := mid // bug? this object contains pointers which are not deep-copied here
```

Comment:
does it mutate that variable?

Reply:
`mid.Base.Path` is the same value on each iteration.  There's something going on here that's not quite right, but it produces the right result.  I think we are rendering the same files multiple times.  I didn't really get to the bottom of it, but that line looked suspicious to me.